test/unit/gateways/cecabank_rest_json_test.rb

```diff
@@ -178,11 +180,52 @@ def test_purchase_without_exemption_type
 
   def test_successful_purchase_with_apple_pay
     stub_comms(@no_encrypted_gateway, :ssl_post) do
-      @no_encrypted_gateway.purchase(@amount, @apple_pay_network_token, @options.merge(xid: 'some_xid'))
+      @no_encrypted_gateway.purchase(@amount, @apple_pay_network_token, @options.merge(xid: XID))
+    end.check_request do |_endpoint, data, _headers|
+      data = JSON.parse(data)
+      params = JSON.parse(Base64.decode64(data['parametros']))
+      common_ap_gp_assertions(
+        params,
+        payment_method: @apple_pay_network_token,
+        wallet_type: 'A',
+        eci: @apple_pay_network_token.eci
+      )
+    end.respond_with(successful_purchase_response)
+  end
+
+  def test_successful_purchase_with_eci_in_three_ds_and_not_apple_pay
+    apple_pay_no_eci = @apple_pay_network_token.dup
+    apple_pay_no_eci.eci = nil
+    stub_comms(@no_encrypted_gateway, :ssl_post) do
+      # Will use eci from three_d_secure instead of payment_method
+      @no_encrypted_gateway.purchase(@amount, apple_pay_no_eci, @options.merge(three_d_secure: @three_d_secure))
+    end.check_request do |_endpoint, data, _headers|
+      data = JSON.parse(data)
+      params = JSON.parse(Base64.decode64(data['parametros']))
+      common_ap_gp_assertions(
```

Comment:
💬 COmment

Seems to be working, but the params doesn't match with the `common_ap_gp_assertions` method signature, am I missing something?